### PR TITLE
time handling fixes

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/basic/time-util.c [hunk 1]
```diff
@@ -185,7 +185,7 @@ usec_t triple_timestamp_by_clock(triple_timestamp *ts, clockid_t clock) {
 usec_t timespec_load(const struct timespec *ts) {
         assert(ts);
 
-        if (ts->tv_sec == (time_t) -1 && ts->tv_nsec == (long) -1)
+        if (ts->tv_sec < 0 && ts->tv_nsec < 0)
```
Comment (keszybz) [inline src/basic/time-util.c]: Hm, what does it mean to have e.g. tv_sec == -1, tv_nsec == 100? Shouldn't this be `||` instead?

File: src/nspawn/nspawn.c [hunk 2]
```diff
@@ -2317,14 +2317,16 @@ static int inner_child(
...
 
-                exec_target = a[0] = (char*) "/lib/systemd/systemd";
+                a[0] = (char*) "/lib/systemd/systemd";
                 execve(a[0], a, env_use);
 
-                exec_target = a[0] = (char*) "/sbin/init";
+                a[0] = (char*) "/sbin/init";
                 execve(a[0], a, env_use);
+
+                exec_target = "/usr/lib/systemd/systemd + /lib/systemd/systemd + /sbin/init";
```
Comment (keszybz) [inline src/nspawn/nspawn.c]: Maybe "/usr/lib/systemd/systemd, /lib/systemd/systemd, /sbin/init"? A bit shorter and easier to understand.

File: src/basic/time-util.c [hunk 3]
```diff
@@ -837,9 +839,14 @@ int parse_timestamp(const char *t, usec_t *usec) {
                 return -EINVAL;
 
         ret = (usec_t) x * USEC_PER_SEC + x_usec;
+        if (ret > USEC_TIMESTAMP_FORMATTABLE_MAX)
+                return -EINVAL;
```
Comment (keszybz) [inline src/basic/time-util.c:844]: Is this check needed, very similar check is right below?

File: src/basic/time-util.c (hunk 1 above)
Comment (poettering) [reply src/basic/time-util.c]: yeah, probably

File: src/basic/time-util.c (hunk 3 above)
Comment (poettering) [reply src/basic/time-util.c:844]: well, each guard one addition. If you permit both additions without individual checks then you might wrap around twice, hiding the overflow...
not that it matters really, but i wanted to be correct here ;-)

File: src/nspawn/nspawn.c (hunk 2 above)
Comment (poettering) [reply src/nspawn/nspawn.c]: saves two bytes, indeed! SO MUCH BYTES! ;-)

**Comment by poettering** [comment]
will shortly force push a fixed version

**Comment by benjarobin** [comment]
@poettering  I create a pull request with the fix : https://github.com/systemd/systemd/pull/5216

**Pull request merged**